scripts/convert_to_csv.py

```diff
@@ -0,0 +1,40 @@
+import pyreadr
+import glob, os
+
+
+errors = []
+
+for fn in glob.glob("../data/rda_files/*.rda"):
+    try:
+        data = pyreadr.read_r(fn, timezone="PST")
+    except Exception as e:
+        errors.append(os.path.basename(fn))
+        continue
+    if not data:
+        errors.append(os.path.basename(fn))
+        continue
+    for df in data.values():
+        if os.path.basename(fn) == "ncdc_normals.rda":
+            df.to_csv(
+                os.path.join(
+                    "..",
+                    "data",
+                    "csv_files",
+                    f"{os.path.splitext(os.path.basename(fn))[0]}.csv.zip",
+                ),
+                index=False,
+                compression="zip",
+            )
+        else:
+            df.to_csv(
+                os.path.join(
+                    "..",
+                    "data",
+                    "csv_files",
+                    f"{os.path.splitext(os.path.basename(fn))[0]}.csv",
+                ),
+                index=False,
+            )

```

Comment:
These two blocks are identical except for the `compression="zip"` part. One approach to consolidating code is to make that argument conditional:
```
if base_name in NEED_ZIP:
    zip_kw = dict(compression="zip")
else: 
    zip_kw = dict()
```
Then you just need a single codepath for the call to `to_csv` that gets the configuration:
```
          df.to_csv(
                os.path.join(
                    "..",
                    "data",
                    "csv_files",
                    f"{os.path.splitext(os.path.basename(fn))[0]}.csv",
                ),
                index=False,
                **zip_kw
```

Reply:
Also personal style pref to take or leave, but if black is expanding vertically hugely this way, I take it as a sign to pull some values out e.g. 
```python
from os.path import join, splitext, basename
```
then
```python
csv_path = f"{splitext(base_name)[0]}.csv
df.to_csv(csv_path, index=False, *zip_kw)
```

Reply:
Last comment :) Although using `os.path` is fine, recent versions of Python have `pathlib` in the std library and that might be worth looking at as a more modern approach